codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientInterface.java

```diff
@@ -169,7 +169,10 @@ public ClassName className() {
 
     private String getJavadoc() {
         return "Service client for accessing " + model.getMetadata().getDescriptiveServiceName() + " asynchronously. This can be "
-               + "created using the static {@link #builder()} method.\n\n" + model.getMetadata().getDocumentation();
+               + "created using the static {@link #builder()} method.\nWe strongly discourage "
+               + "configuring main I/O threads to run future-completion objects. This could "
+               + "lead to deadlock as the SDK may perform blocking calls in some cases."
+               + "\n\n" + model.getMetadata().getDocumentation();
```

Comment:
This may be a bit confusing for customers who are not aware of future completion config. Suggesting the following:
```
... This can be created using the static `builder()` method.  
<p>
The async client performs non-blocking I/O when configured with any {@link SdkAsyncHttpClient} 
supported in the SDK. However, full non-blocking is not guaranteed as the async client may perform 
blocking calls in some cases such as credentials fetching, endpoint discovery as part of the async API call.
```